### PR TITLE
test: add otelc integration test

### PR DESCRIPTION
**Pull request by fabriziosestito**
## Description
Adds otel collector integration test.

The otelc container is started using  `testcontainers`, and configured with file exporters.
We can then read and parse the file contents and assert that everything is fine.

Fixes: #855

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -65,6 +65,8 @@ rstest = "0.21"
 tempfile = "3.10.1"
 tower = { version = "0.4", features = ["util"] }
 http-body-util = "0.1.1"
+testcontainers = { version = "0.21.1", features = ["watchdog"] }
+backon = "0.4.4"
```
Comment (flavio) [inline Cargo.toml:69]: Nice addition! I think I can refactor the unit test of the certificate reload to make use of it

File: tests/integration_test.rs [hunk 2]
```diff
@@ -16,12 +24,21 @@ use policy_evaluator::{
...
+use tempfile::NamedTempFile;
+use testcontainers::{
+    core::{Mount, WaitFor},
+    runners::AsyncRunner,
+    GenericImage, ImageExt,
+};
 use tower::ServiceExt;
 
 use crate::common::default_test_config;
+use backon::{ExponentialBuilder, Retryable};
```
Comment (flavio) [inline tests/integration_test.rs:41]: nitpick: should be moved upper, like around line 38

File: tests/integration_test.rs [hunk 3]
```diff
@@ -625,7 +642,8 @@ async fn test_detect_certificate_rotation() {
     let key_file = certs_dir.path().join("policy-server-key.pem");
 
     let hostname1 = "cert1.example.com";
-    let tls_data1 = create_cert(&hostname1);
+    let var_name = &hostname1;
+    let tls_data1 = create_cert(var_name);
```
Comment (flavio) [inline tests/integration_test.rs:646]: I'm a bit puzzled about this change, nothing else was touched inside of the test. Do we need it?

File: tests/integration_test.rs [hunk 4]
```diff
@@ -679,3 +697,105 @@ async fn test_detect_certificate_rotation() {
...
+    assert_eq!(spans["spans"][0]["name"], "policy_log");
+    assert_eq!(spans["spans"][1]["name"], "validate_settings");
+
+    otelc.stop().await.unwrap();
+}
+
+async fn parse_exporter_output(
+    exporter_output_file: &File,
+) -> serde_json::Result<serde_json::Value> {
+    // let exporter_output_file = File::open(exporter_output_file_path).unwrap();
```
Comment (flavio) [inline tests/integration_test.rs:793]: Can be removed

**Review by flavio — CHANGES_REQUESTED** [review]
Looks really neat! I left some minor comments, once addressed this is ready to be merged

File: tests/integration_test.rs (hunk 3 above)
Comment (fabriziosestito) [reply tests/integration_test.rs:646]: the linter was complaining, i just applied the suggestion

Comment (fabriziosestito) [reply tests/integration_test.rs:646]: you are right, it seems i've applied the wrong code action, let me revert it

**Review by viccuad — APPROVED** [review]
LGTM!

File: Cargo.toml (hunk 1 above)
Comment (jvanz) [reply Cargo.toml:69]: Indeed! Very cool

**Review by jvanz — APPROVED** [review]

**Review by flavio — APPROVED** [review]

**Pull request merged**